### ISSUE TITLE
If I just want to use the pre-trained model, should I have to have a pc with Navidia graphics card?

### ISSUE THREAD

**Issue by lc8631058** [facebookresearch/deepmask#101]
Hi, guys, I have a Macbook pro with Inter graphics card, when I try to use Quick Start, there are some errors like that:
`congcong (master) deepmask $ th computeProposals.lua $DEEPMASK/pretrained/deepmask
/Users/congcong/torch/install/bin/luajit: /Users/congcong/torch/install/share/lua/5.1/trepl/init.lua:389: module 'cutorch' not found:No LuaRocks module found for cutorch
	no field package.preload['cutorch']
	no file '/Users/congcong/.luarocks/share/lua/5.1/cutorch.lua'
	no file '/Users/congcong/.luarocks/share/lua/5.1/cutorch/init.lua'
	no file '/Users/congcong/torch/install/share/lua/5.1/cutorch.lua'
	no file '/Users/congcong/torch/install/share/lua/5.1/cutorch/init.lua'
	no file './cutorch.lua'
	no file '/Users/congcong/torch/install/share/luajit-2.1.0-beta1/cutorch.lua'
	no file '/usr/local/share/lua/5.1/cutorch.lua'
	no file '/usr/local/share/lua/5.1/cutorch/init.lua'
	no file '/Users/congcong/.luarocks/lib/lua/5.1/cutorch.so'
	no file '/Users/congcong/torch/install/lib/lua/5.1/cutorch.so'
	no file '/Users/congcong/torch/install/lib/cutorch.dylib'
	no file './cutorch.so'
	no file '/usr/local/lib/lua/5.1/cutorch.so'
	no file '/usr/local/lib/lua/5.1/loadall.so'
stack traceback:
	[C]: in function 'error'
	/Users/congcong/torch/install/share/lua/5.1/trepl/init.lua:389: in function 'require'
	computeProposals.lua:11: in main chunk
	[C]: in function 'dofile'
	...cong/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:150: in main chunk
	[C]: at 0x0105a08350`

It seems like I have no cutorch, but when I try to use `luarocks install cutorch` to install it, error comes again, is that a matter of graphics card? thanks a lot:

`Installing https://raw.githubusercontent.com/torch/rocks/master/cutorch-scm-1.rockspec...
Using https://raw.githubusercontent.com/torch/rocks/master/cutorch-scm-1.rockspec... switching to 'build' mode
Cloning into 'cutorch'...
remote: Counting objects: 227, done.
remote: Compressing objects: 100% (181/181), done.
remote: Total 227 (delta 63), reused 90 (delta 44), pack-reused 0
Receiving objects: 100% (227/227), 233.18 KiB | 200.00 KiB/s, done.
Resolving deltas: 100% (63/63), done.
Warning: unmatched variable LUALIB

jopts=$(getconf _NPROCESSORS_CONF)

echo "Building on $jopts cores"
cmake -E make_directory build && cd build && cmake .. -DLUALIB= -DLUA_INCDIR=/Users/congcong/torch/install/include -DCMAKE_CXX_FLAGS=${CMAKE_CXX_FLAGS} -DCMAKE_BUILD_TYPE=Release -DCMAKE_PREFIX_PATH="/Users/congcong/torch/install/bin/.." -DCMAKE_INSTALL_PREFIX="/Users/congcong/torch/install/lib/luarocks/rocks/cutorch/scm-1" && make -j$jopts install

Building on 4 cores
-- The C compiler identification is AppleClang 8.0.0.8000042
-- The CXX compiler identification is AppleClang 8.0.0.8000042
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Torch7 in /Users/congcong/torch/install
CMake Error at /usr/local/Cellar/cmake/3.8.2/share/cmake/Modules/FindCUDA.cmake:642 (message):
  Specify CUDA_TOOLKIT_ROOT_DIR
Call Stack (most recent call first):
  CMakeLists.txt:7 (FIND_PACKAGE)

-- Configuring incomplete, errors occurred!
See also "/tmp/luarocks_cutorch-scm-1-2925/cutorch/build/CMakeFiles/CMakeOutput.log".

Error: Build error: Failed building.`

**Comment by RJ722** [comment]
@lc8631058 I am not sure yet!

But I think `cutorch` needs `CUDA` and you would most probably need a GPU for it. 😕 

Please have a look at the [official docs](http://docs.nvidia.com/cuda/#axzz3km0bJe3r ) for finding out whether or not does your computer can install CUDA?

**Comment by ROZBEH** [comment]
Not really, I was able to install it on MAC OS.

**Comment by RJ722** [comment]
> Not really, I was able to install it on MAC OS.

It is definitely possible on Mac OS X, it depends on whether or not your Macbook has a graphics card. 😉

**Comment by ROZBEH** [comment]
Is it possible to install it on MAC with intel Graphic card?

**Comment by RJ722** [comment]
Nope! :-( becuase `cutorch` needs `CUDA` and you would need an NVIDIA chip for it.

**Comment by ROZBEH** [comment]
Thanks @RJ722 . Is there any solution to this?

**Comment by RJ722** [comment]
I was not able to get it working on my machine - I didn't try much though!

Once cutorch is installed, you would be good to go. So, look for a way to install CUDA/cutorch on machines not having an NVIDIA chip! :)
